tests/gocase/integration/slotimport/slotimport_test.go

```diff
@@ -0,0 +1,144 @@
+	clusterNodes = fmt.Sprintf("%s\n%s 127.0.0.1 %d slave %s", clusterNodes, srvBID, srvB.Port(), srvAID)
+
+	require.NoError(t, rdbA.Do(ctx, "clusterx", "SETNODES", clusterNodes, "1").Err())
+	require.NoError(t, rdbB.Do(ctx, "clusterx", "SETNODES", clusterNodes, "1").Err())
+
+	t.Run("IMPORT - Can't import slot to slave", func(t *testing.T) {
+		time.Sleep(100 * time.Millisecond)
+		require.ErrorContains(t, rdbB.Do(ctx, "cluster", "import", 1, 0).Err(), "Slave can't import slot")
+	})
+}
+
+func TestImportedServer(t *testing.T) {
+	ctx := context.Background()
+
+	srv := util.StartServer(t, map[string]string{"cluster-enabled": "yes"})
+	defer func() { srv.Close() }()
+	rdb := srv.NewClient()
+	defer func() { require.NoError(t, rdb.Close()) }()
+	srvID := "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx00"
+	clusterNodes := fmt.Sprintf("%s 127.0.0.1 %d master - 0-16383", srvID, srv.Port())
+	require.NoError(t, rdb.Do(ctx, "clusterx", "SETNODEID", srvID).Err())
+	require.NoError(t, rdb.Do(ctx, "clusterx", "SETNODES", clusterNodes, "1").Err())
+
+	t.Run("IMPORT - error slot", func(t *testing.T) {
+		require.ErrorContains(t, rdb.Do(ctx, "cluster", "import", -1, 0).Err(), "Slot is out of range")
+		require.ErrorContains(t, rdb.Do(ctx, "cluster", "import", 16384, 0).Err(), "Slot is out of range")
+	})
+
+	t.Run("IMPORT - slot with error state", func(t *testing.T) {
+		require.ErrorContains(t, rdb.Do(ctx, "cluster", "import", 1, 4).Err(), "Invalid import state")
+		require.ErrorContains(t, rdb.Do(ctx, "cluster", "import", 1, -3).Err(), "Invalid import state")
+	})
+
+	t.Run("IMPORT - slot states in right order", func(t *testing.T) {
+		slotNum := 1
+		slotKey := util.GetKeyWithSlotNum(slotNum)
+
+		// import start
+		require.Equal(t, "OK", rdb.Do(ctx, "cluster", "import", slotNum, 0).Val())
+		require.NoError(t, rdb.Set(ctx, slotKey, "slot1", 0).Err())
+		require.Equal(t, "slot1", rdb.Get(ctx, slotKey).Val())
+		clusterInfo := rdb.ClusterInfo(ctx).Val()
+		require.Contains(t, clusterInfo, "importing_slot: 1")
+		require.Contains(t, clusterInfo, "import_state: start")
+
+		// import success
+		require.Equal(t, "OK", rdb.Do(ctx, "cluster", "import", slotNum, 1).Val())
+		clusterInfo = rdb.ClusterInfo(ctx).Val()
+		require.Contains(t, clusterInfo, "importing_slot: 1")
+		require.Contains(t, clusterInfo, "import_state: success")
+
+		time.Sleep(50 * time.Millisecond)
+		require.Equal(t, "slot1", rdb.Get(ctx, slotKey).Val())
+	})
+
+	t.Run("IMPORT - slot state 'error'", func(t *testing.T) {
+		slotNum := 10
+		slotKey := util.GetKeyWithSlotNum(slotNum)
+
+		require.Equal(t, "OK", rdb.Do(ctx, "cluster", "import", slotNum, 0).Val())
+		require.NoError(t, rdb.Set(ctx, slotKey, "slot10_again", 0).Err())
+		require.Equal(t, "slot10_again", rdb.Get(ctx, slotKey).Val())
+
+		// import error
+		require.Equal(t, "OK", rdb.Do(ctx, "cluster", "import", slotNum, 2).Val())
+		time.Sleep(50 * time.Millisecond)
```

Comment:
I guess these sleeps will generate unstable tests. We can keep an eye on these cases later.